include/oxen/quic/connection.hpp

```diff
@@ -129,9 +129,9 @@ namespace oxen::quic
         template <oxenc::basic_char Char>
         void send_datagram(std::vector<Char>&& buf)
         {
-            send_datagram(
-                    std::basic_string_view<Char>{buf.data(), buf.size()},
-                    std::make_shared<std::vector<Char>>(std::move(buf)));
+            auto keep_alive = std::make_shared<std::vector<Char>>(std::move(buf));
+            std::basic_string_view<Char> view{keep_alive->data(), keep_alive->size()};
+            send_datagram(view, std::move(keep_alive));
```

Comment:
Though this change won't hurt, it isn't necessary, either.  From https://en.cppreference.com/w/cpp/container/vector/vector: 

> Notes
> 
> After container move construction (overload (8)), references, pointers, and iterators (other than the end iterator) to other remain valid, but refer to elements that are now in *this

i.e. the move transfers ownership, but does not invalidate the pointer to the data.  (This isn't the case for string, however, because of small-string optimization).

Reply:
This is the exact bug we encountered (the three of us) in the tunnel binaries. I'm just lifting the fix from the WIP PR